src/lib.rs

```diff
@@ -924,7 +923,7 @@ mod test {
     fn test_clone() -> Result<()> {
         let owned_con = checked_memory_handle();
         {
-            let cloned_con = owned_con.clone();
+            let cloned_con = owned_con.try_clone().unwrap();
             cloned_con.execute_batch("PRAGMA VERSION")?;
         }
         owned_con.close().unwrap();
```

Comment:
I looked into the implementation of `close` -- Isn't there a bug in the usage of `owned` and `close()` here?

The initial object has `owned = true` and the cloned one has `owned = false`. If we drop the original before dropping the clone (opposite order of this test) wouldn't the original actually close the DuckDB file and leave the clone with a dangling reference?

It seems like we need a shared reference counter to close the DB correctly, rather than this `owned` boolean.

Reply:
It could be a problem, could you write a test about it?